spec/amber/validations/optional_rule_spec.cr

```diff
@@ -21,6 +22,20 @@ module Amber::Validators
         end
 
         rule.apply(params).should be_true
+        rule.value.should_not be_nil
+        rule.value.should eq "val"
+      end
+
+      it "allows an empty value" do
+        params = params_builder("field=")
+        error_message = "You must provide this field"
+        rule = OptionalRule.new("field", error_message) do
+          true
+        end
+
+        rule.apply(params).should be_true
+        rule.value.should_not be_nil
+        rule.value.should eq ""
```

Comment:
`should_not be_nil` is redundant with `should eq ""` both here and above.

Reply:
Agreed, I think I just put that in there to contrast with the original example, but that's not a great reason.